workflow/cron/operator.go

```diff
@@ -178,6 +184,89 @@ func (woc *cronWfOperationCtx) patch(ctx context.Context, patch map[string]inter
 	}
 }
 
+// TODO: refactor shouldExecute in steps.go
+func shouldExecute(when string) (bool, error) {
+	if when == "" {
+		return true, nil
+	}
+	expression, err := govaluate.NewEvaluableExpression(when)
```

Comment:
oh? I thought we were using `expr` here given `govaluate`'s deprecation??

Reply:
Not yet, we are using the exact logic for how When is evaluated in hooks as well.

Reply:
See follow-ups [below](https://github.com/argoproj/argo-workflows/pull/13474#issuecomment-2327932625)